django/db/models/base.py

```diff
@@ -889,8 +889,9 @@ def _save_table(self, raw=False, cls=None, force_insert=False,
 
             returning_fields = meta.db_returning_fields
             results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
-            for result, field in zip(results, returning_fields):
-                setattr(self, field.attname, result)
+            if results:
```

Comment:
Is this necessary? :thinking:  `SQLInsertCompiler.execute_sql()` always return a list.

Reply:
The next line's `results[0]` fails if it's empty